edalize/cocotb.py

```diff
@@ -0,0 +1,141 @@
+import logging
+import os.path
+from edalize.edatool import Edatool
+
+logger = logging.getLogger(__name__)
+
+# This is a list of simulators where VHDL libraries are supported in cocotb
+sims_with_vhdl_lib_support = ['ghdl']
+# Minor hack to deal with verilog include directories. Patching cocotb to do the
+# simulator specifics maybe better long term. Basing this off cocotb
+# axi-lite-slave example Makefile.
+verilog_include_prefix = {'icarus': '-I', 'default': '+incdir+'}
+
+
+class Cocotb(Edatool):
+
+    argtypes = []
+
+    makefile_template = 'cocotb-makefile.j2'
+
+    @classmethod
+    def get_doc(cls, api_ver):
+        if api_ver == 0:
+            return {'description': 'cocotb - see <https://docs.cocotb.org/> for additional documentation on options',
+                    'members': [
+                        {'name': 'sim',
+                         'type': 'String',
+                         'desc': 'The simulator for Cocotb to use (required)'},
+                        {'name': 'module',
+                         'type': 'String',
+                         'desc': 'Comma-separated list of the Python modules to search for test functions (required)'},
+                        {'name': 'toplevel_lang',
+                         'type': 'String',
+                         'desc': 'Language of top level HDL module (required)'},
+                        {'name': 'rtl_library',
+                         'type': 'String',
+                         'desc': 'VHDL library for top level entity'},
+                        {'name': 'gui',
+                         'type': 'String',
+                         'desc': 'Set to 1 to enable GUI mode in simulator'},
+                        {'name': 'waves',
+                         'type': 'String',
+                         'desc': 'Set to 1 to enable wave traces dump'},
+                        {'name': 'compile_args',
+                         'type': 'String',
+                         'desc': 'Any arguments or flags to pass to the compile stage of the simulation'},
+                        {'name': 'sim_args',
+                         'type': 'String',
+                         'desc': 'Any arguments or flags to pass to the execution of the compiled simulation'},
+                        {'name': 'run_args',
+                         'type': 'String',
+                         'desc': 'Any argument to be passed to the “first” invocation of a simulator that runs via a TCL script'},
+                        {'name': 'extra_args',
+                         'type': 'String',
+                         'desc': 'Passed to both the compile and execute phases of simulators with'
+                         + ' two rules, or passed to the single compile and run command for simulators'
+                         + ' which don’t have a distinct compilation stage'},
```

Comment:
nit: faster to use parse-time concatenation:
```suggestion
                         'desc': 'Passed to both the compile and execute phases of simulators with '
                                 'two rules, or passed to the single compile and run command for simulators '
                                 'which don’t have a distinct compilation stage'},
```